test/suites/chaos/zonal_failure_test.go

```diff
@@ -0,0 +1,489 @@
+			env.EventuallyExpectHealthyPodCountWithTimeout(5*time.Minute, selector, numPods)
+
+			By("Cleaning up test resources")
+			env.ExpectDeleted(dep)
+			env.ExpectExperimentTemplateDeleted(templateID)
+
+			Eventually(func(g Gomega) {
+				nodeList := &corev1.NodeList{}
+				g.Expect(env.Client.List(env.Context, nodeList, client.HasLabels{coretest.DiscoveryLabel})).To(Succeed())
+				for _, node := range nodeList.Items {
+					g.Expect(env.Client.Delete(env.Context, &node)).To(Succeed())
+				}
+				nodeClaimList := &karpv1.NodeClaimList{}
+				g.Expect(env.Client.List(env.Context, nodeClaimList, client.HasLabels{coretest.DiscoveryLabel})).To(Succeed())
+				for _, nodeClaim := range nodeClaimList.Items {
+					g.Expect(env.Client.Delete(env.Context, &nodeClaim)).To(Succeed())
+				}
+
+				g.Eventually(func() int {
+					tempNodeList := &corev1.NodeList{}
+					g.Expect(env.Client.List(env.Context, tempNodeList, client.HasLabels{coretest.DiscoveryLabel})).To(Succeed())
+					return len(tempNodeList.Items)
+				}, 2*time.Minute, 10*time.Second).Should(BeZero())
+
+				describeInstancesOutput, err := env.EC2API.DescribeInstances(env.Context, &ec2.DescribeInstancesInput{
+					Filters: []ec2types.Filter{
+						{
+							Name:   aws.String(fmt.Sprintf("tag:%s", coretest.DiscoveryLabel)),
+							Values: []string{env.ClusterName},
+						},
+					},
+				})
+				g.Expect(err).NotTo(HaveOccurred())
+
+				var instanceIDs []string
+				for _, reservation := range describeInstancesOutput.Reservations {
+					for _, instance := range reservation.Instances {
+						if instance.InstanceId != nil {
+							instanceIDs = append(instanceIDs, *instance.InstanceId)
+						}
+					}
+				}
+
+				if len(instanceIDs) > 0 {
+					fmt.Printf("Terminating %d remaining EC2 instances\n", len(instanceIDs))
+					_, err := env.EC2API.TerminateInstances(env.Context, &ec2.TerminateInstancesInput{
+						InstanceIds: instanceIDs,
+					})
+					g.Expect(awserrors.IgnoreNotFound(err)).NotTo(HaveOccurred())
+				}
+			}, 5*time.Minute).Should(Succeed())
+		},
+		Entry("complete failure", "100", "complete-failure-app", "complete AZ failure"),
+		Entry("moderate failure", "50", "moderate-failure-app", "moderate AZ failures"),
+		Entry("minor failure", "25", "minor-failure-app", "minor AZ failures"),
+	)
+})
+
+func createExperimentTemplate(ctx context.Context, env *awsenv.Environment, targetAZ string, instances []ec2types.Instance, failurePercentage string) string {
+	// Filter instances to only include those in the target AZ
+	var targetInstances []string
+	for _, instance := range instances {
+		if instance.InstanceId != nil && instance.Placement != nil && instance.Placement.AvailabilityZone != nil {
+			targetInstances = append(targetInstances, *instance.InstanceId)
+		}
+	}
+
+	subnetARNs := getSubnetsInAZ(ctx, env, targetAZ)
+	By(fmt.Sprintf("Found %d subnets in AZ %s", len(subnetARNs), targetAZ))
+
+	template := &fis.CreateExperimentTemplateInput{
```

Comment:
What do you think about adding the experiment resource to our resource monitoring dashboard?